tsdb/meta.go

```diff
@@ -1810,15 +1830,26 @@ func MarshalTags(tags map[string]string) []byte {
 // TagKeys returns a list of the measurement's tag names.
 func (m *Measurement) TagKeys() []string {
 	m.mu.RLock()
-	defer m.mu.RUnlock()
 	keys := make([]string, 0, len(m.seriesByTagKeyValue))
 	for k := range m.seriesByTagKeyValue {
 		keys = append(keys, k)
 	}
+	m.mu.RUnlock()
 	sort.Strings(keys)
 	return keys
 }
 
+func (m *Measurement) TagKeysBytes() [][]byte {
```

Comment:
I'm not sure this method is useful. It copies the string contents, rather than referencing the strings that already exist as map keys. The only usage of this that I could find is to check the cardinality (which takes a `[]byte`). Perhaps `Cardinality` should take a `string` and add `CardinalityBytes` to match other similar methods we have.